api/v1/logicalvolume_types.go

```diff
@@ -18,6 +18,17 @@ type LogicalVolumeSpec struct {
 	NodeName    string            `json:"nodeName"`
 	Size        resource.Quantity `json:"size"`
 	DeviceClass string            `json:"deviceClass,omitempty"`
+
+	// 'source' specifies the logicalvolume name of the source; if present.
+	// +kubebuilder:validation:Optional
+	Source string `json:"source"`
```

Comment:
This should be omitEmpty as well. It is not required for a regular logicalvolume.